src/Language/PureScript/Errors.hs

```diff
@@ -12,108 +12,378 @@
+errorCode UnspecifiedSkolemScope        = "UnspecifiedSkolemScope"
+errorCode (TypesDoNotUnify _ _)         = "TypesDoNotUnify"
+errorCode (KindsDoNotUnify _ _)         = "KindsDoNotUnify"
+errorCode (ConstrainedTypeUnified _ _)  = "ConstrainedTypeUnified"
+errorCode (OverlappingInstances _ _ _)  = "OverlappingInstances"
+errorCode (NoInstanceFound _ _)         = "NoInstanceFound"
+errorCode (DuplicateLabel _ _)          = "DuplicateLabel"
+errorCode (DuplicateValueDeclaration _) = "DuplicateValueDeclaration"
+errorCode (ArgListLengthsDiffer _)      = "ArgListLengthsDiffer"
+errorCode (OverlappingArgNames _)       = "OverlappingArgNames"
+errorCode (MissingClassMember _)        = "MissingClassMember"
+errorCode (ExpectedType _)              = "ExpectedType"
+errorCode (IncorrectConstructorArity _) = "IncorrectConstructorArity"
+errorCode SubsumptionCheckFailed        = "SubsumptionCheckFailed"
+errorCode (ExprDoesNotHaveType _ _)     = "ExprDoesNotHaveType"
+errorCode (PropertyIsMissing _ _)       = "PropertyIsMissing"
+errorCode (ErrorUnifyingTypes _ _ _)    = "ErrorUnifyingTypes"
+errorCode (CannotApplyFunction _ _)     = "CannotApplyFunction"
+errorCode TypeSynonymInstance           = "TypeSynonymInstance"
+errorCode InvalidNewtype                = "InvalidNewtype"
+errorCode (InvalidInstanceHead _)       = "InvalidInstanceHead"
+errorCode (TransitiveExportError _ _)   = "TransitiveExportError"
+errorCode (NotYetDefined _ e)           = errorCode e
+errorCode (ErrorInExpression _ e)       = errorCode e
+errorCode (ErrorInModule _ e)           = errorCode e
+errorCode (ErrorInInstance _ _ e)       = errorCode e
+errorCode (ErrorInSubsumption _ _ e)    = errorCode e
+errorCode (ErrorCheckingType _ _ e)     = errorCode e
+errorCode (ErrorCheckingKind _ e)       = errorCode e
+errorCode (ErrorInferringType _ e)      = errorCode e
+errorCode (ErrorInApplication _ _ _ e)  = errorCode e
+errorCode (ErrorInDataConstructor _ e)  = errorCode e
+errorCode (ErrorInTypeConstructor _ e)  = errorCode e
+errorCode (ErrorInBindingGroup _ e)     = errorCode e
+errorCode (ErrorInDataBindingGroup e)   = errorCode e
+errorCode (ErrorInTypeSynonym _ e)      = errorCode e
+errorCode (ErrorInValueDeclaration _ e) = errorCode e
+errorCode (ErrorInForeignImport _ e)    = errorCode e
+errorCode (PositionedError _ e)         = errorCode e
 
--- TODO: Remove strMsg, the IsString instance, and unnecessary
--- OverloadedStrings pragmas. See #745
--- | Create an ErrorStack from a string
-strMsg :: String -> ErrorStack
-strMsg s = ErrorStack [CompileError s Nothing Nothing]
-
-instance IsString ErrorStack where
-  fromString = strMsg
+-- |
+-- Pretty print an ErrorMessage
+--
+prettyPrintErrorMessage :: ErrorMessage -> String
+prettyPrintErrorMessage e =
+  go e ++ "\n\nSee " ++ wikiUri ++ " for more information, or to contribute content related to this error."
+  where
+  wikiUri :: String
+  wikiUri = "http://wiki.purescript.org/Error-Code-" ++ errorCode e
```

Comment:
Actually, the `wiki.purescript.org` link only redirects over HTTP. I'll need to take a look at the domain settings to see if I can do anything about that. Otherwise, I'll link to `github.com`.